spyderlib/widgets/file_switcher.py

```diff
@@ -0,0 +1,415 @@
+              border: 1px solid grey;
+              padding:0px;
+              border-radius: 2px;
+              background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
+                  stop: 0 #f6f7fa, stop: 1 #dadbde);
+            }
+            """
+        self.button_help.setStyleSheet(style)
+        help_str = _("""
+            Press <b>Enter</b> to switch files.<br>
+            Or <b>Esc</b> to cancel.<br>
+            Type to filter filenames.<br>
+            Use <b>:number</b> to go to a line,<br>
+            e.g. 'main:42'.<br>
+            Press <b>Ctrl+W</b> to close current tab.
+            """)
+        self.button_help.setToolTip(help_str)
+        edit_layout.addWidget(self.button_help)
+
+        self.listwidget = QListWidget(self)
+        self.listwidget.setItemDelegate(HTMLDelegate(self))
+        self.listwidget.itemSelectionChanged\
+                       .connect(self.item_selection_changed)
+        self.listwidget.itemActivated.connect(self.handle_edit_file)
+        self.original_path = None
+        self.original_line_num = None
+        self.line_num = -1
+        self.line_num_modified_for_path = None
+
+        self.edit_line.connect(self.handle_edit_line)
+        self.rejected.connect(self.handle_rejection)
+        new_shortcut("Ctrl+W", self, lambda:
+                     self.close_file.emit(
+                         self.filtered_index_to_full(
+                             self.listwidget.currentRow())))
+
+        vlayout = QVBoxLayout()
+        vlayout.addLayout(edit_layout)
+        vlayout.addWidget(self.listwidget)
+        self.setLayout(vlayout)
+
+        self.edit.selectAll()
+        self.edit.setFocus()
+        geo = parent.geometry()
+        width = min(300, 0.8*geo.width())
+        self.listwidget.setMinimumWidth(width)
+
+        left = parent.geometry().width()/2 - width/2
+        top = 0
+        while parent:
+            geo = parent.geometry()
+            top += geo.top()
+            left += geo.left()
+            parent = parent.parent()
+        self.move(left,
+                  top + self.tabs.tabBar().geometry().height() + 1)
+        # Note: the +1px on the top is a hack
+
+    def filtered_index_to_full(self, idx):
+        # Note we assume idx is valid and the two mappings are valid
+        return self.full_index_to_path.index(self.filtered_index_to_path[idx])
+
+    def handle_up_down(self, up_down):
+        row = self.listwidget.currentRow() + up_down
+        if 0 <= row < self.listwidget.count():
+            self.listwidget.setCurrentRow(row)
+
+    def handle_rejection(self):
+        self.edit_line.emit(-1)  # reset line number for current tab, if change
+        if self.original_path is not None:
+            self.edit_file.emit(self.full_index_to_path
+                                .index(self.original_path))
+
+    def handle_edit_file(self):
+        row = self.listwidget.currentRow()
+        if self.listwidget.count() > 0 and row >= 0:
```

Comment:
`if self.listwidget.count() and row >= 0:` is sufficient.
Also, can `row >= 0` be False if self.listwidget.count() > 0 ?